tlatools/org.lamport.tlatools/test/pcal/parser/PcalParserTranslator.java

```diff
@@ -0,0 +1,184 @@
+package pcal.parser;
+
+import java.util.Arrays;
+import java.util.Vector;
+
+import util.AstNode;
+import util.AstNode.Kind;
+
+import pcal.AST;
+import pcal.PcalCharReader;
+import pcal.TLAtoPCalMapping;
+import pcal.TLAExpr;
+import pcal.TLAToken;
+import pcal.ParseAlgorithm;
+import pcal.PcalParams;
+import pcal.exception.ParseAlgorithmException;
+
+import org.junit.Assert;
+
+/**
+ * Utilities for parsing PlusCal and converting it to a form compatible with
+ * the unified TLA+ syntax parser test corpus.
+ */
+public class PcalParserTranslator {
+	
+	/**
+	 * Simple data class holding info about the start of a PlusCal block.
+	 */
+	private static class PcalStart {
+		/**
+		 * The line on which the PlusCal block starts.
+		 */
+		int line = 0;
+		
+		/**
+		 * The column in which the PlusCal block starts, directly after the
+		 * --algorithm or --fair algorithm tokens.
+		 */
+		int column = 0;
+		
+		/**
+		 * Whether the PlusCal block is a fair algorithm.
+		 */
+		boolean isFair = false;
+		
+		/**
+		 * Finds the start of the PlusCal block.
+		 * 
+		 * @param input Text containing PlusCal block, split into lines.
+		 */
+		public PcalStart(Vector<String> input) {
+			final String fairStart = "--fair";
+			final String unfairStart = "--algorithm";
+			for (int i = 0; i < input.size(); i++) {
+				String line = input.get(i);
+				int keywordStart = line.indexOf(fairStart);
+				if (-1 == keywordStart) {
+					keywordStart = line.indexOf(unfairStart);
+					if (-1 == keywordStart) {
+						continue;
+					} else {
+						this.line = i;
+						this.column = keywordStart + unfairStart.length();
+						this.isFair = false;
+						break;
+					}
+				} else {
+					this.line = i;
+					this.column = keywordStart + fairStart.length();
+					this.isFair = true;
```

Comment:
Nit: all of these class fields can be `final`. That's good style, even for a test-only class, and especially since these fields are non-private. For code like this that contains loops, you'll need to satisfy Java's final-field-initialization-checks by introducing local variables (`line`, `column`, and `isFair`) and writing those into the object as the last few lines in the constructor:
```
this.line = line;
this.column = column;
this.isFair = isFair;
```

I'm not sure how robust this class needs to be; you may also need to detect the case where no begin token was found.